repository/gorm/repository_test.go

```diff
@@ -51,13 +52,23 @@ func TestMain(m *testing.M) {
 		common2,
 		common3,
 	}
-	if err := migration.CreateDatabasesIfNotExists("mysql", fmt.Sprintf("%s:%s@tcp(%s:%s)/?charset=utf8mb4&parseTime=true", user, pass, host, port), dbPrefix, dbs...); err != nil {
+	config := &driverMysql.Config{
+		User:                 user,
+		Passwd:               pass,
+		Net:                  "tcp",
+		Addr:                 fmt.Sprintf("%s:%s", host, port),
+		Collation:            "utf8mb4_general_ci",
+		ParseTime:            true,
+		AllowNativePasswords: true,
+	}
+	if err := migration.CreateDatabasesIfNotExists("mysql", config.FormatDSN(), dbPrefix, dbs...); err != nil {
 		panic(err)
 	}
 
 	for _, key := range dbs {
+		config.DBName = fmt.Sprintf("%s%s", dbPrefix, key)
 		engine, err := gorm.Open(mysql.New(mysql.Config{
-			DSN: fmt.Sprintf("%s:%s@tcp(%s:%s)/%s?charset=utf8mb4&parseTime=true", user, pass, host, port, fmt.Sprintf("%s%s", dbPrefix, key)),
+			DSNConfig: config,
```

Comment:
結構細かいですが、

この実装は `config.DBName = fmt.Sprintf("%s%s", dbPrefix, key)` と `config` 構造体の元の値を直接変更してるけど、以下の様に元の値を直接変更するのではなくコピーしてから変更した方が、コードの意図が明確になって、さらに他の部分への影響を防ぐことでメンテナンス上でバグが起きにくくなりそうです！

```suggestion
		dbConfig := *config
		dbConfig.DBName = fmt.Sprintf("%s%s", dbPrefix, key)
		engine, err := gorm.Open(mysql.New(mysql.Config{
			DSNConfig: &dbConfig,
```